flax/linen/module.py

```diff
@@ -96,8 +96,8 @@ def is_module_tree(in_tree: Any) -> bool:
   # reject trivial pytrees, {}, [], (), etc.
   if not tree_util.tree_leaves(in_tree):
     return False
-  reduce_fn = lambda prev, cur: prev and isinstance(cur, Module)
-  return jax.tree_util.tree_reduce(reduce_fn, in_tree, True)
+  reduce_fn = lambda prev, cur: prev or isinstance(cur, Module)
```

Comment:
Is this function really useful still? It seems like a useless extra traversal of each pytree?
Why not just tree_map every pytree in assignment and register the leaves that are modules?

Reply:
yeah this seems super dumb now, I'll try to remove this nonsense tomorrow.

Reply:
ok, I cleaned all this up